src/compiler/checker.ts

```diff
@@ -22560,6 +22560,28 @@ namespace ts {
             return filterType(type, t => (getTypeFacts(t) & include) !== 0);
         }
 
+        function truthyTypeFilter(type: Type, truthiness: boolean): Type | undefined {
+            if (!truthiness) {
+                if (type.flags & TypeFlags.BigInt) {
+                    return zeroBigIntType;
+                }
+                else if (type.flags & TypeFlags.Boolean) {
+                    return falseType;
+                }
+                else if (type.flags & TypeFlags.String) {

```

Comment:
There probably needs to be a `TemplateString` and `StringMapping` case now, too, since both type kinds could possibly contain the empty string.